ext/ffi/ffi.c

```diff
@@ -3756,12 +3756,8 @@ ZEND_METHOD(FFI, cast) /* {{{ */
 
 	ZEND_FFI_VALIDATE_API_RESTRICTION();
 	ZEND_PARSE_PARAMETERS_START(2, 2)
-		if (Z_TYPE_P(EX_VAR_NUM(0)) == IS_STRING) {
-			Z_PARAM_STR(type_def)
-		} else {
-			Z_PARAM_OBJECT_OF_CLASS(ztype, zend_ffi_ctype_ce)
-		}
-		Z_PARAM_ZVAL(zv);
+		Z_PARAM_STR_OR_OBJ_OF_CLASS(type_def, ztype, zend_ffi_ctype_ce)
+		Z_PARAM_ZVAL(zv)
```

Comment:
Can we use `Z_PARAM_OBJECT_OF_CLASS` instead? Or is there any problem it causes together with `prefer-ref`?

Reply:
Yeah, that won't work.

Reply:
Also, looking at the implementation it accepts more than just `FFI\CData`, can also be null, int or string.

Reply:
Oh, I see it now!